domain/src/main/java/org/oppia/android/domain/devoptions/ShowAllHintsAndSolutionHelper.kt

```diff
@@ -0,0 +1,19 @@
+package org.oppia.android.domain.devoptions
+
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Monitor to identify whether all hints and solution should be shown. */
+@Singleton
+class ShowAllHintsAndSolutionHelper @Inject constructor() {
+
+  private var showAllHintsAndSolution = false
+
+  /** Returns [showAllHintsAndSolution] indicating whether showing all hints and solution feature is enabled. */
+  fun getShowAllHintsAndSolution(): Boolean = showAllHintsAndSolution
+
+  /** Sets [showAllHintsAndSolution]. */
+  fun setShowAllHintsAndSolution(isEnabled: Boolean) {
+    showAllHintsAndSolution = isEnabled
```

Comment:
Is it possible to access the dev options menu while within an exploration? If so, this doesn't actually notify the handler impl that the state has changed (meaning there will be a delayed update). Is that okay?

Reply:
No, it is not possible to access the menu while within an exploration.